pkg/controller/statefulset/stateful_set_control.go

```diff
@@ -208,8 +208,11 @@ func (ssc *defaultStatefulSetControl) getStatefulSetRevisions(
 			return nil, nil, err
 		}
 	} else {
+		if set.Status.CollisionCount == nil {
+			set.Status.CollisionCount = new(int64)
```

Comment:
I noticed that `updateStatefulSet` makes a copy of `StatefulSetStatus` and `updateStatefulSetStatus` makes a copy of the `StatefulSet`.  Given that, we probably should have `getStatefulSetRevisions` return a collision count and then pass it into `updateStatefulSet` so it gets picked up. What's your thoughts?